source/features/add-timezone-to-hovercard.js

```
@@ -0,0 +1,103 @@
+	const locationRequest = await fetch(
+		`${locationAPI}${encodeURIComponent(location)}`
+	);
+	const locationResponse = await locationRequest.json();
+
+	try {
+		const {lat, lng} = locationResponse.results[0].geometry.location;
+		const timezoneRequest = await fetch(createTimezoneRequestURL(lat, lng));
+		const timezoneResponse = await timezoneRequest.json();
+
+		return timezoneResponse;
+	} catch (error) {
+		return null;
+	}
+}
+
+function traverseNodelist(node, positionList, i) {
+	// NodeLists are live and must be converted to arrays
+	if (i === positionList.length - 1) {
+		return Array.from(node[positionList[i]].childNodes);
+	}
+
+	// Traverse down the tree to the desired div
+	return traverseNodelist(
+		Array.from(node[positionList[i]].childNodes),
+		positionList,
+		i + 1
+	);
+}
+
+function getLocationFromHovercard(node) {
+	// These are locations of NodeLists nested in other NodeLists
+	const nodePositions = [0, 0, 5, 3, 7];
+	const target = traverseNodelist(node, nodePositions, 0);
+	return target[2].textContent.trim();
+}
+
+export default async function () {
+	const observer = new MutationObserver(async mutationsList => {
+		let hoverCardMutation;
+		for (const mutation of mutationsList) {
+			if (mutation.type === 'childList' && mutation.addedNodes.length > 0) {
+				hoverCardMutation = mutation;
+				break;
+			}
+		}
+
+		try {
+			const location = getLocationFromHovercard(
+				Array.from(hoverCardMutation.addedNodes)
+			);
+
+			// There will only be a single hoverCard at any time
+			const hoverCard = select('div.mt-2.text-gray.text-small');
+			const timeZone = await getTimezone(location);
+
+			if (timeZone) {
+				const {timeZoneId} = timeZone;
+				const time = moment.tz(moment(), timeZoneId).format('HH:mm');
+				const abbr = moment.tz(timeZoneId).zoneAbbr();
+				hoverCard.append(clockIcon());
+				hoverCard.append(domify(`${time} ${abbr}`));
+			}
+		} catch (e) {
+			// Silently fail when there are too many API requests
+		}
```

I don't think we need suppress the errors. If it's not working we should probably show why.

---

It will fail when either the `timezoneAPI` or the `locationAPI`request fails due to too many requests, but the requests often succeed on the next mouseover. How do you want to handle this?

---

I'd let it go to the console for now. Perhaps it can use [`onetime`](https://www.npmjs.com/package/onetime) to show the error in the console once per load at least, so it doesn't fail completely silently _and_ doesn't spam the console with errors.

But @sindresorhus perhaps can give better advice on expected error handling.